# Description of `Schools_stats` table

In [2]:
import pandas as pd

In [4]:
df = pd.read_csv('../tables_creation/final_tables/school_stats.csv')
df.head()

EDRPOU  KOATUU_2020_reg eotype eolevel  eosize  teachstuff  \
0  26244113.0                5  urban   I-III     NaN        44.0   
1  26244107.0                5  urban   I-III     NaN        59.0   
2  26244136.0                5  urban   I-III     NaN        82.0   
3  26244099.0                5  urban   I-III     NaN        61.0   
4  23064971.0                5  urban   I-III     NaN        70.0   

   nonteachstuff  teachstuffretage  pupils  classes     opex opexplan hub  \
0           20.0               8.0     539       20   9915.2    11123  no   
1           18.0               8.0     840       30  13692.2  13758.8  no   
2           23.0              11.0     981       36  16184.7  17254.2  no   
3           22.0               6.0     854       32  14321.9  14598.6  no   
4           23.0               8.0     864       33  15061.0  16011.5  no   

   year  
0  2019  
1  2019  
2  2019  
3  2019  
4  2019

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18898 entries, 0 to 18897
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   EDRPOU            18897 non-null  float64
 1   KOATUU_2020_reg   18898 non-null  int64  
 2   eotype            18898 non-null  object 
 3   eolevel           18898 non-null  object 
 4   eosize            0 non-null      float64
 5   teachstuff        18898 non-null  float64
 6   nonteachstuff     18898 non-null  float64
 7   teachstuffretage  18898 non-null  float64
 8   pupils            18898 non-null  int64  
 9   classes           18898 non-null  int64  
 10  opex              18898 non-null  float64
 11  opexplan          18898 non-null  object 
 12  hub               10641 non-null  object 
 13  year              18898 non-null  int64  
dtypes: float64(6), int64(4), object(4)
memory usage: 2.0+ MB


In [6]:
# print unique values for each column
for col in df.columns:
    print('Unique length in ', col+':\t', len(df[col].unique()))

Unique length in  EDRPOU:	 9732
Unique length in  KOATUU_2020_reg:	 25
Unique length in  eotype:	 2
Unique length in  eolevel:	 3
Unique length in  eosize:	 1
Unique length in  teachstuff:	 776
Unique length in  nonteachstuff:	 386
Unique length in  teachstuffretage:	 209
Unique length in  pupils:	 1565
Unique length in  classes:	 79
Unique length in  opex:	 18151
Unique length in  opexplan:	 18239
Unique length in  hub:	 3
Unique length in  year:	 2


Features:
- EDRPOU (unique code of the legal entity or physical person): 9732 unique codes of schools.
- KOATUU_2020_reg (region level of KOATUU code): 25 unique codes denoting regions of school locations (the first two digits of the full KAOTUU code of the location).
- eotype (type of school's settlement). Range of values:
    - rural
    - urban
- eolevel (level of the school; I is elementary school, II is middle school, III is high school; only schools which have III of education have graduates who want to enter universities, and take tests). Range of values:
    - I-III
    - II-III
    - III
- teachstuff (number of teaching stuff). Range of values:
    - 2-215 (integer)
- nonteachstuff (number of non-teaching stuff). Range of values:
    - 0-165 (integer)
- teachstuffretage (number of teaching stuff of retirement age). Range of values:
    - 0-54 (integer)
- pupils (number of pupils). Range of values:
    - 5-2477 (integer)
- classes (number of classes). Range of values:
    - 0-98 (integer)
- opex (expenses for the operation of the school, current year, thousands UAH). Range of values:
    - 247.787-48330.2 (float)
- opexplan (planned expenses for the operation of the school, next year, thousands UAH). Range of values:
    - 0: no information about planned expanses
    - 56.794-83878.7 (float)
- hub (is the school a hub school). Range of values:
    - N/A: no info
    - yes
    - no
- year (year of the statistics info for the school). Range of values:
    - 2019-2020 (integer)